lib/net.js

```diff
@@ -846,12 +846,16 @@ function connect(self, address, port, addressType, localAddress, localPort) {
 }
 
 
-// Check that the port number is not NaN when coerced to a number,
-// is an integer and that it falls within the legal range of port numbers.
+// Check that port is undefined, a number or a string.
```

Comment:
It could be split into two functions: One that checks the type and one that does the coercion to number (`port | 0`) and throws if not valid.